drivers/dma/dma_mcux_lpc.c

```diff
@@ -544,7 +556,14 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 			assert(block_config->dest_address ==
 			       ROUND_UP(block_config->dest_address, width));
 
-			if (dma_mcux_lpc_queue_descriptors(data, block_config, src_inc, dst_inc)) {
+			if (block_config->next_block == NULL) {
+				/* This is the last block. Enable callback if not enabled. */
+				if (!complete_callback) {
+					complete_callback = true;
```

Comment:
dont think the `if` is needed here

Reply:
This is still needed. The previous time this was set was for the case where there is only 1 DMA block.

Reply:
for some reason, I did not see the `!`, and thought it was redundant

Reply:
@mmahadevan108 if you are setting it to true unconditionally why does there need to be a check of if it is false

Reply:
This is done only for the case when we have not additional blocks in the queue. Hence it does not apply for the code further.

Reply:
Thanks, I have made this change.